disnake/guild.py

```diff
@@ -3350,6 +3352,57 @@ async def query_members(
             self, query=query, limit=limit, user_ids=user_ids, presences=presences, cache=cache
         )
 
+    async def search_members(
+        self,
+        query: str,
+        *,
+        limit: int = 1,
+        cache: bool = True,
+    ):
+        """|coro|
+
+        Retrieves members that belong to this guild whose username or nickname starts with
+        the query given.
+
+        Note that unlike :func:`query_members`, this is not a websocket operation.
+
+        See also :func:`query_members`.
+
+        .. versionadded:: 2.5
+
+        Parameters
+        -----------
+        query: Optional[:class:`str`]
+            The string that the usernames or nicknames start with.
+        limit: :class:`int`
+            The maximum number of members to send back. This must be
+            a number between 1 and 1000.
+        cache: :class:`bool`
+            Whether to cache the members internally. This makes operations
+            such as :meth:`get_member` work for those that matched.
+
+        Raises
+        -------
+        ValueError
+            Invalid parameters were passed to the function
+
+        Returns
+        --------
+        List[:class:`Member`]
+            The list of members that have matched the query.
+        """
+        if query == "":
+            raise ValueError("Cannot pass empty query string.")
+        limit = min(1000, limit or 1)
```

Comment:
```suggestion
        limit = min(1000, limit)
```

Reply:
Its not possible to request 0 members, hence this `or`